src/FieldWarning/Assets/Ingame/UI/SelectionManager.cs

```diff
@@ -0,0 +1,200 @@
+﻿/**
+ * Copyright (c) 2017-present, PFW Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+using System.Collections.Generic;
+using UnityEngine;
+
+using Assets.Model.Game;
+using static Assets.Ingame.UI.UIManagerBehaviour;
+using System.Linq;
+
+using static Assets.Ingame.UI.Constants;
+
+namespace Assets.Ingame.UI
+{
+    public class SelectionManager : MonoBehaviour
+    {
+        public List<PlatoonBehaviour> Selection { get; private set; }
+
+        private Vector3 _mouseStart;
+        private Vector3 _mouseEnd;
+        private Texture2D _texture;
+        private Texture2D _borderTexture;
+        private Color _selectionBoxColor = Color.red;
+        private bool _active;
+
+        private ClickManager _clickManager;
+
+        private GameSession _session;
+        public GameSession Session {
+            get {
+                return _session;
+            }
+
+            set {
+                if (_session == null)
+                    _session = value;
+            }
+        }
+
+        public void Awake()
+        {
+            Selection = new List<PlatoonBehaviour>();
+            _clickManager = new ClickManager(0, MOUSE_DRAG_THRESHOLD, StartBoxSelection, OnSelectShortClick, EndDrag, UpdateBoxSelection);
```

Comment:
Do we have to pass the Constant into ClickManager via the constructor?

Reply:
No, good catch.